packages/suite/src/components/wallet/Fees/components/CustomFee.tsx

```diff
@@ -87,26 +91,105 @@ const SetDefaultLimit = ({ onClick }: { onClick: () => void }) => (
 const FEE_PER_UNIT = 'feePerUnit';
 const FEE_LIMIT = 'feeLimit';
 
-const FeeCustom = ({
-    account: { networkType },
+type FormMethods = UseFormMethods<{
+    selectedFee?: FeeLevel['label'];
+    feePerUnit?: string;
+    feeLimit?: string;
+    estimatedFeeLimit?: string;
+}>;
+
+interface CustomFeeProps {
+    networkType: Account['networkType'];
+    feeInfo: FeeInfo;
+    errors: FormMethods['errors'];
+    register: (rules?: TypedValidationRules) => (ref: any) => void;
+    getValues: FormMethods['getValues'];
+    setValue: FormMethods['setValue'];
+    changeFeePerUnit?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    changeFeeLimit?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const CustomFee = ({
+    networkType,
     feeInfo,
     errors,
     register,
     getValues,
     setValue,
     changeFeePerUnit,
     changeFeeLimit,
-}: Props) => {
+}: CustomFeeProps) => {
     const { maxFee, minFee } = feeInfo;
-    const feePerUnitValue = getValues(FEE_PER_UNIT);
-    const feePerUnitError = errors.feePerUnit;
 
-    const useFeeLimit = networkType === 'ethereum';
+    const feePerUnitValue = getValues(FEE_PER_UNIT);
     const feeLimitValue = getValues(FEE_LIMIT);
     const estimatedFeeLimit = getValues('estimatedFeeLimit') || ETH_DEFAULT_GAS_LIMIT;
+
+    const feePerUnitError = errors.feePerUnit;
     const feeLimitError = errors.feeLimit;
+
+    const useFeeLimit = networkType === 'ethereum';
     const feeLimitDisabled = false;
 
+    const validateFeeLimit = (value: string) => {
+        const feeBig = new BigNumber(value);
+
+        if (feeBig.isNaN()) {
+            return 'CUSTOM_FEE_IS_NOT_NUMBER';
+        }
+
+        // allow decimals in ETH since GWEI is not a satoshi
+        if (networkType !== 'ethereum' && networkType !== 'bitcoin' && !isInteger(value)) {
+            return 'CUSTOM_FEE_IS_NOT_INTEGER';
+        }
+
+        if (feeBig.lt(estimatedFeeLimit)) {
+            return (
+                <SetDefaultLimit
+                    onClick={() => {
+                        setValue(FEE_LIMIT, estimatedFeeLimit, {
+                            shouldValidate: true,
+                        });
+                    }}
+                />
+            );
+        }
+    };
+
+    const validateFee = (value: string) => {
+        const feeBig = new BigNumber(value);
+
+        if (feeBig.isNaN()) {
+            return 'CUSTOM_FEE_IS_NOT_NUMBER';
+        }
+
+        // allow decimals in ETH since GWEI is not a satoshi
```

Comment:
```suggestion
        // allow decimals in ETH and BTC since GWEI and SAT are not Lovelace 🌷
```
Just kidding, but comment in `validateFee` and `validateFeeLimit` should be changed

Reply:
I left the ETH one becase the reasoning for that one hasn't changed but I am not sure what the reasoning for BTC should be :D The task mentions 2 decimals but I was thinking we want to allow mSats, however those are 3 decimals, right?